lib/Controller/SubsonicController.php

```diff
@@ -1067,17 +1071,49 @@ protected function deleteBookmark(string $id) : array {
 	/**
 	 * @SubsonicAPI
 	 */
-	protected function getPlayQueue() : array {
-		// TODO: not supported yet
-		return ['playQueue' => []];
+	protected function getPlayQueue() : Response {
+		$defaultResponse = [
+			'entry' => [],
+			'changedBy' => ''
+		];
+		$playQueue = json_decode($this->configManager->getUserValue($this->user(), $this->appName, 'play_queue', 'false'), true) ?: $defaultResponse;
+
+		foreach ($playQueue['entry'] as &$entry) {
+			[$type, $id] = self::parseEntityId($entry);
+			if ($type === 'track') {
+				$entry = $this->trackToApi(
+					$this->trackBusinessLayer->find($id, $this->user())
+				);
+			} else if ($type === 'podcast_episode') {
+				$entry = $this->podcastEpisodeBusinessLayer->find($id, $this->user())->toSubsonicApi();
+			}
+		}
```

Comment:
I wonder about this section. If an `$entry` somehow doesn't match either `"track"` or `"podcast_episode"`, it'll be left in the response as an invalid element. Are there any other types of media that might end up saved in `savePlayQueue`? If this is a reasonable concern (and I dunno if it is), I can solve it by sanitizing the input on `savePlayQueue` or by generating a new array of just the valid items from `$playQueue['entry']` in `getPlayQueue`

Reply:
That's a good question. The third kind of media playable on the Subsonic clients is a radio station. I'm not certain if any client puts those to the play queue, though. The API would allow also video playback, but this is not supported in the Music app (at least for now or in the near future).

But yeah, it would probably be safer to construct the list so that only the supported types of entities would be retained.